lib/ReactViews/Clipboard.tsx

```diff
@@ -1,86 +1,77 @@
-import clipboard from "clipboard";
-import { ReactElement, FC, useEffect, useState } from "react";
+import { FC, useEffect, useState, type ReactNode } from "react";
 import { useTranslation } from "react-i18next";
 import styled from "styled-components";
 import Box from "../Styled/Box";
 import Button from "../Styled/Button";
+import Icon, { StyledIcon } from "../Styled/Icon";
 import { verticalAlign } from "../Styled/mixins";
 import Spacing from "../Styled/Spacing";
-import Icon, { StyledIcon } from "../Styled/Icon";
 
 enum CopyStatus {
   Success,
   Error,
-  NotCopiedOrWaiting // Copy button hasn't been clicked or clipboard.js hasn't copied the data yet
+  Default
 }
 
 interface ClipboardProps {
-  id: string;
-  source: ReactElement;
+  source: ReactNode;
   theme: "dark" | "light";
   rounded?: boolean;
   text?: string;
   onCopy?: (contents: string) => void;
 }
 
 const Clipboard: FC<ClipboardProps> = (props) => {
-  const { id, source, theme, rounded } = props;
+  const { source, theme, rounded, text, onCopy } = props;
   const { t } = useTranslation();
-  const [status, setStatus] = useState<CopyStatus>(
-    CopyStatus.NotCopiedOrWaiting
-  );
-  useEffect(() => {
-    // Setup clipboard.js and show a tooltip on copy success or error for 3s
-    const clipboardBtn = new clipboard(`.btn-copy-${id}`);
-    let timerId: ReturnType<typeof setTimeout> | null = null;
-    function removeTimeout() {
-      if (timerId !== null) {
-        clearTimeout(timerId);
-        timerId = null;
+  const [status, setStatus] = useState<CopyStatus>(CopyStatus.Default);
+
+  const handleCopy = async () => {
+    try {
+      if (text) {
+        await navigator.clipboard.writeText(text);
+        setStatus(CopyStatus.Success);
+        if (onCopy) onCopy(text);
+      } else {
+        setStatus(CopyStatus.Error);
       }
+    } catch {
+      setStatus(CopyStatus.Error);
     }
-    function resetTooltipLater() {
-      removeTimeout();
-      timerId = setTimeout(() => {
-        setStatus(CopyStatus.NotCopiedOrWaiting);
+  };
+
+  useEffect(() => {
+    if (status === CopyStatus.Success || status === CopyStatus.Error) {
+      const timer = setTimeout(() => {
+        setStatus(CopyStatus.Default);
       }, 3000);
+
+      return () => clearTimeout(timer);
     }
-    clipboardBtn.on("success", (evt) => {
-      props.onCopy?.(evt.text);
-      setStatus(CopyStatus.Success);
-      resetTooltipLater();
-    });
-    clipboardBtn.on("error", () => {
-      setStatus(CopyStatus.Error);
-      resetTooltipLater();
-    });
-    return function cleanup() {
-      removeTimeout();
-      clipboardBtn.destroy();
-    };
-    /* eslint-disable-next-line react-hooks/exhaustive-deps */
-  }, [id]);
+  }, [status]);
 
   const isLightTheme = theme === "light";
+  const canCopy = !!navigator.clipboard;
+
   return (
-    <ClipboardDiv>
+    <ClipboardDiv rounded={!!rounded}>
       <Box>
```

Comment:
@zoran995 - thanks for making the change but I think `rounded` needs to be applied to the `<Box>` instead. Otherwise it will clip the status message and looks a bit odd:

![image](https://github.com/user-attachments/assets/ffbc7b2c-5443-413c-9dbe-81cf16c05e9a)

Reply:
I have updated the PR. I was testing on CI and didn't realise I had forgotten to push the commits to both remotes, so they were out of sync. Later, I also realised that we can't use `overlay: hidden` as it will break the focus outline, so I adapted the solution to work without breaking it. This should be ready for final review